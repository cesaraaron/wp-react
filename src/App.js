import React from 'react'
import { Route } from 'react-router-dom'
import Posts from './Home'
import Single from './Single'
import Category from './Category'

const App = () => (
  <div>
    <Route exact path="/" component={Posts} />
    <Route path="/page/:pageNumber" component={Posts} />
    <Route exact path="/:slug" component={Single} />
    <Route path="/category/:slug" component={Category} />
  </div>
)

export default App
