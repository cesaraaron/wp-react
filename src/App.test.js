import React from 'react'
import App from './App'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from './reducers'

describe('<App />', () => {
  const store = createStore(rootReducer)
  store.dispatch = () => {}

  it('should render without crashing', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
