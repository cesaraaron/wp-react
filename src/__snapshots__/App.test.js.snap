// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render without crashing 1`] = `
<div>
  <div>
    <div>
      <div />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="text"
      />
      <div />
    </div>
    <br />
    <div />
  </div>
</div>
`;
